instrumentation/github.com/go-kit/kit/otelkit/config.go

```diff
@@ -106,3 +109,11 @@ func WithAttributeGetter(fn func(ctx context.Context) []attribute.KeyValue) Opti
 		o.GetAttributes = fn
 	})
 }
+
+// WithSpanKind sets the span kind for spans created by this middleware.
+// If this option is not provided, then trace.SpanKindServer is used.
+func WithSpanKind(kind trace.SpanKind) Option {
```

Comment:
I think that only two values of `trace.SpanKind` are valid:

- `trace.SpanKindServer`
- `trace.SpanKindConsumer`


Can we change and add `WithSpanKindServer()` and `WithSpanKindConsumer()` options instead?

Credits to @Aneurysm9

Reply:
I've been using `SpanKindInternal` on my endpoints and instead putting server/consumer on the transport, as it's up to the transport whether the endpoint is called as part of a server or consumer. Changing this to just those two options somewhat limits the flexibility here.

Reply:
I am not following. Why would you set to `Internal`? 

>  as it's up to the transport whether the endpoint is called as part of a server or consumer

How would you modify the span kind? If it would be possible then you would not even need this functionality.

However, now I see that go-kit transport/endpoint can also be a "consumer" or "client" so probably it is better to keep as it is. Approving again  😉